src/screencast/wlr_screencast.c

```diff
@@ -325,28 +344,40 @@ static void wlr_registry_handle_add(void *data, struct wl_registry *reg,
 		uint32_t id, const char *interface, uint32_t ver) {
 	struct xdpw_screencast_context *ctx = data;
 
+	logprint(DEBUG, "wlroots: interface to register %s  (Version: %u)",interface, ver);
 	if (!strcmp(interface, wl_output_interface.name)) {
 		struct xdpw_wlr_output *output = malloc(sizeof(*output));
 
 		output->id = id;
-		output->output = wl_registry_bind(reg, id, &wl_output_interface, 1);
+		logprint(DEBUG, "wlroots: |-- registered to interface %s (Version %u)", interface, WL_OUTPUT_VERSION);
+		output->output = wl_registry_bind(reg, id, &wl_output_interface, WL_OUTPUT_VERSION);
 
 		wl_output_add_listener(output->output, &wlr_output_listener, output);
 		wl_list_insert(&ctx->output_list, &output->link);
 	}
 
 	if (!strcmp(interface, zwlr_screencopy_manager_v1_interface.name)) {
+		uint32_t version = ver;
+		if (SC_MANAGER_VERSION < ver) {
+			version = SC_MANAGER_VERSION;
+		} else if (ver < SC_MANAGER_VERSION_MIN) {
+			version = SC_MANAGER_VERSION_MIN;
+		}
+		ctx->state->screencast_version = version;
```

Comment:
No need to store this information. We can get the version back via `zwlr_screencopy_manager_v1_get_version`.

Reply:
We need this information later for every frame (https://github.com/columbarius/xdg-desktop-portal-wlr/blob/screencopy-interface-v3/src/screencast/wlr_screencast.c#L169), so would prefer to safe this and not request it every time. It also doesn't change between different screencasts.

Reply:
`zwlr_screencopy_manager_v1_get_version` is not a request, it's just a getter.

Reply:
Ahh thanks. Changed it.